packages/core/data-transfer/src/strapi/queries/link.ts

```diff
@@ -302,9 +302,7 @@ export const createLinkQuery = (strapi: Core.Strapi, trx?: Knex.Transaction) =>
         assignOrderColumns();
         const qb = connection.insert(payload).into(addSchema(joinTable.name));
         if (trx) {
-          await trx.transaction(async (nestedTrx) => {
-            await qb.transacting(nestedTrx);
-          });
+          await qb.transacting(trx);
```

Comment:
We don't need to create a new transaction, we're already in one for the data transfer. I think this may have been causing some connection pool issues.

This isn't a strictly necessary fix for this PR, but I've included it here because the amount of manual testing on data transfer issues is high, and I'd rather do it just once.

Reply:
yeah indeed not needed, it's shouldn't have been causing issues though, if am not wrong the logic of the transaction will already ignore this nested one because it's inside one already